gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -146,6 +151,19 @@ static void addJobStatusToStateStore(org.apache.gobblin.configuration.State jobS
     String storeName = jobStatusStoreName(flowGroup, flowName);
     String tableName = jobStatusTableName(flowExecutionId, jobGroup, jobName);
 
+    List<org.apache.gobblin.configuration.State> states = stateStore.getAll(storeName, tableName);
```

Comment:
Can we avoid duplicate stateStore.getAll() call, as we are doing one in mergedProperties() as well?

Reply:
Good point, addressed.